apps/src/lib/kits/maker/boards/microBit/MicroBitBoard.js

```diff
@@ -95,34 +89,46 @@ export default class MicroBitBoard extends EventEmitter {
    * @returns {Promise<void>}
    */
   checkExpectedFirmware() {
-    return Promise.resolve()
-      .then(() => this.openSerialPort())
-      .then(serialPort => this.boardClient_.connectBoard(serialPort))
-      .then(() => {
-        // Delay for 0.25 seconds to ensure we have time to receive the firmware version.
-        return delayPromise(250);
-      })
-      .then(() => {
-        if (
-          this.boardClient_.firmwareVersion.includes(MICROBIT_FIRMWARE_VERSION)
-        ) {
-          return Promise.resolve();
-        } else {
-          if (this.boardClient_.firmwareVersion === '') {
-            // Log if we were not able to determine the firmware version in time.
-            firehoseClient.putRecord({
-              study: 'maker-toolkit',
-              study_group: 'microbit',
-              event: 'firmwareVersionTimeout'
-            });
-            console.warn(
-              'Firmware version not detected in time. Try refreshing the page.'
-            );
+    if (!isWebSerialPort(this.port)) {
+      // maker app pathway
+      return Promise.resolve()
+        .then(() => this.openSerialPort())
+        .then(serialPort => this.boardClient_.connectBoard(serialPort))
+        .then(() => {
+          // Delay for 0.25 seconds to ensure we have time to receive the firmware version.
+          return delayPromise(250);
+        })
+        .then(() => {
+          if (
+            this.boardClient_.firmwareVersion.includes(
+              MICROBIT_FIRMWARE_VERSION
+            )
+          ) {
+            return Promise.resolve();
+          } else {
+            if (this.boardClient_.firmwareVersion === '') {
+              // Log if we were not able to determine the firmware version in time.
+              firehoseClient.putRecord({
+                study: 'maker-toolkit',
+                study_group: 'microbit',
+                event: 'firmwareVersionTimeout'
+              });
+              console.warn(
+                'Firmware version not detected in time. Try refreshing the page.'
+              );
+            }
+            return Promise.reject('Incorrect firmware detected');
           }
-          return Promise.reject('Incorrect firmware detected');
-        }
-      })
-      .catch(err => Promise.reject(err));
+        })
+        .catch(err => Promise.reject(err));
+    } else {
+      // webserial pathway
+      return this.openSerialPortWebSerial(this.port)
```

Comment:
Note that `this.port` is assigned in the constructor above which is passed from `detectCorrectFirmware` in `SetupChecker`.
`SetupChecker` is used in `SetupChecklist` and `SetupChecklist` is imported in `SetupInstructions`. It is within `SetupInstructions.jsx` on line 74-76 that `navigator.serial.requestPort` is called and the value of `webSerialPort` is set.